From e75a7652a072e5dbb2ceafdcde6824a2db33941d Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 2 Oct 2018 06:40:27 +0200
Subject: [PATCH 1/1] efi_loader: error handling in read_console()

getc() might return an error code. Avoid an incorrect converison to
Unicode.

This addresses CoverityScan CID 184087.

Reported-by: Tom Rini <trini@konsulko.com>
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/charset.c | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/lib/charset.c b/lib/charset.c
index 0cede9b60b..10557b9e75 100644
--- a/lib/charset.c
+++ b/lib/charset.c
@@ -97,12 +97,17 @@ static u8 read_string(void *data)
 /**
  * read_console() - read byte from console
  *
- * @src		- not used, needed to match interface
- * Return:	- byte read
+ * @data	- not used, needed to match interface
+ * Return:	- byte read or 0 on error
  */
 static u8 read_console(void *data)
 {
-	return getc();
+	int ch;
+
+	ch = getc();
+	if (ch < 0)
+		ch = 0;
+	return ch;
 }
 
 int console_read_unicode(s32 *code)
-- 
2.19.0

